src/FunFunFactory.php

```diff
@@ -135,17 +136,35 @@ private function newPimple(): \Pimple {
 			] );
 		} );
 
+		$pimple['translator'] = $pimple->share( function() {
+			$translationFactory = new TranslationFactory();
+			$loaders = [
+				'json' => $translationFactory->newJsonLoader()
+			];
+			$locale = 'de_DE'; // In the future, this could be configured somehow
+			$translator = $translationFactory->create( $loaders, $locale );
+			$translator->addResource( 'json', __DIR__ . '/../app/translations/messages.de_DE.json', $locale );
+			$translator->addResource( 'json', __DIR__ . '/../app/translations/validations.de_DE.json', $locale, 'validations' );
+			return $translator;
+		} );
+
 		$pimple['twig_factory'] = $pimple->share( function () {
 			return new TwigFactory( $this->config['twig'] );
 		} );
 
 		$pimple['twig'] = $pimple->share( function( $pimple ) {
+			/** @var TwigFactory $twigFactory */
```

Comment:
The rest of the code avoids the issue you fixed here by creating a getter, such as the `getTranslator` you introduced here.